### DISCUSSION TITLE
Species range model V2.4 - V2, Jan 2024

### DISCUSSION THREAD

**Discussion by Punkley** [mcguirepr89/BirdNET-Pi#1105] [Ideas]
Hi [BirdNET-Analyzer](https://github.com/kahst/BirdNET-Analyzer) has upgraded to  V2.4 - V2, could we get birdnet-pi updated please :)

**Comment by Svardsten53** [comment]
I agree with Punkley, it would be nice too have this update implemented in BirdNET-Pi. I hope **ehpersonal38** can take a look at this as you mentioned when we earlier discussed future support. Thanks!

**Comment by pddpauw** [comment]
wouldn't it be as simple as copying the new V2 model from BirdNET-Analyzer ([here](https://github.com/kahst/BirdNET-Analyzer/blob/main/checkpoints/V2.4/BirdNET_GLOBAL_6K_V2.4_MData_Model_V2_FP16.tflite)) to the folder in BirdNET-Pi ([here](https://github.com/mcguirepr89/BirdNET-Pi/tree/main/model)), and do the correct redirection in the server.py, species.py and birdnet.conf file?    If yes, can easily give it a try!

**Reply by Oripy** [reply]
Have you tried it in the end? Any news on this model update?

**Reply by pddpauw** [reply]
included in my repo for Pi5

**Comment by PeterMescher** [comment]
Well, since there were no label changes (it was just a geographic range update, not an update to species recognized, or the underlying AI model) I pulled down the MData file and simply renamed it (removed the "V2" from the MData file) and it appears to work just fine.

If I was feeling industrious, I could have dug into the code to figure out where the MData file was referenced (and changed it there, instead of simply changing the filename), but since I'm lazy, I didn't.

**Reply by Oripy** [reply]
Which file(s?) exactly did you got from [BirdNet-Analyzer](https://github.com/kahst/BirdNET-Analyzer)?

**Reply by PeterMescher** [reply]
https://github.com/kahst/BirdNET-Analyzer/blob/70545748fca25bd8ac2081ae27a5c6f530a1ec49/checkpoints/V2.4/BirdNET_GLOBAL_6K_V2.4_MData_Model_V2_FP16.tflite

Just remove the "_V2" from the filename, and plop it right in pi/BirdNET-pi/models, right where the existing MData file is.

**Reply by Oripy** [reply]
Perfect, that did the trick! Thanks

**Reply by Svardsten53** [reply]
I have also done the modification on my two installations. Thank you!

**Reply by krummrey** [reply]
It really is that simple, rename and replace. 👍

**Reply by CGKYoung** [reply]
Does the "preview species list" work for you? I get the image below. 

<img width="960" alt="image" src="https://github.com/mcguirepr89/BirdNET-Pi/assets/150294456/6bd0d8cb-fec1-4e08-aca3-2c9904c0c8de">

**Reply by PeterMescher** [reply]
That list works fine for me.  (Though I'm using an LXC fork, not the main branch)

**Comment by stevedee** [comment]
The explanation for this v2 version (which is twice the size of the current one for BirdNET-Pi) is here: https://github.com/kahst/BirdNET-Analyzer/discussions/234

I'll update my 2nd system with these changes and test when time permits.

**Comment by kilbyjmichael** [comment]
Is the owner of the repo still active? Does anyone else have commit rights?

**Reply by MueJosh** [reply]
> Is the owner of the repo still active? Does anyone else have commit rights?

Check 

[#1003](https://github.com/mcguirepr89/BirdNET-Pi/discussions/1003)
